### PR TITLE
Added time dilation tracking

### PR DESCRIPTION
**Pull request by DamianX**
## What Does This PR Do
Added SStime_track, taking parts of it from TG.
AA wanted it and I wanted to look into how this was done.

## Why It's Good For The Game
You have more insight into time slowing down, when it does.

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Labels for GBP bounty, technically a feature

File: code/controllers/subsystem/time_track.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
+SUBSYSTEM_DEF(time_track)
+	name = "Time Tracking"
+	wait = 100
+	init_order = INIT_ORDER_TIME_TRACK
+	runlevels = RUNLEVEL_LOBBY | RUNLEVELS_DEFAULT
+	flags = SS_NO_INIT
```
Comment (AffectedArc07) [inline code/controllers/subsystem/time_track.dm:6]: Any reason youre adding an init order when it has `SS_NO_INIT`?

File: code/controllers/subsystem/time_track.dm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	var/first_run = TRUE
+
+	var/last_tick_realtime = 0
+	var/last_tick_byond_time = 0
+	var/last_tick_tickcount = 0
+
+/datum/controller/subsystem/time_track/fire()
+	var/current_realtime = REALTIMEOFDAY
+	var/current_byondtime = world.time
+	var/current_tickcount = world.time/world.tick_lag
```
Comment (AffectedArc07) [inline code/controllers/subsystem/time_track.dm:22]:
```suggestion
	var/current_tickcount = world.time / world.tick_lag
```

File: code/controllers/subsystem/time_track.dm (hunk 1 above)
Comment (DamianX) [reply code/controllers/subsystem/time_track.dm:6]: The original had one, I forgot to remove the init order

File: code/controllers/subsystem/time_track.dm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	last_tick_byond_time = current_byondtime
+	last_tick_tickcount = current_tickcount
+
+/datum/controller/subsystem/time_track/get_metrics()
+	. = ..()
+	.["custom"] = list(
+		"time_dilation_current" = "[time_dilation_current]",
+		"time_dilation_avg_fast" = "[time_dilation_avg_fast]",
+		"time_dilation_avg" = "[time_dilation_avg]",
+		"time_dilation_avg_slow" = "[time_dilation_avg_slow]",
```
[the remark this answers is not in the document]
Comment (DamianX) [reply code/controllers/subsystem/time_track.dm:43]: I don't remember when they get turned into scientific notation. I tested this and got nothing conclusive:
```
var/a = 1.12345678912345
world.log << a
world.log << "[a]"
world.log << num2text(a)
world.log << json_encode(a)
world.log << json_encode("[a]")
world.log << json_encode(num2text(a))
```

```
1.12346
1.12346
1.12346
1.12346
"1.12346"
"1.12346"
```

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**